clients/drcachesim/common/options.cpp

```diff
@@ -333,3 +333,15 @@ droption_t<bool> op_reuse_verify_skip
  "Verifies every skip list-calculated reuse distance with a full list walk. "
  "This incurs significant additional overhead.  This option is only available "
  "in debug builds.");
+
+// XXX: replace function return address with function callstack
+droption_t<std::string> op_record_function
+(DROPTION_SCOPE_FRONTEND, "record_function", DROPTION_FLAG_ACCUMULATE, ";",
+ std::string("malloc|0|1") + ";" + std::string("tc_malloc|1|1") + ";" +
```

Comment:
Hmm, so by default every trace will include the values for these 3 functions -- not sure about that for a couple of reasons:

1) It seems incomplete: someone mapping addresses to data structures or sthg will want at least free() as well, if not calloc and new, etc.  Yet there's no comment here or anything in the option doc saying these default values are in-progress and not final (yes, a pet peeve of mine: I like prototype or incremental code to clearly mark itself).

2) Should this be in *all* address traces by default: I guess compared to the trace overhead the wrapping overhead should be minimal and the trace size expansion should be minimal (should we measure?)  But how would we arrange to have a heap tracing option: some kind of alias convenience option `-record_heap` which turns into `-record_function malloc...free...calloc` (or it could set up the recording internally after the option parsing).

Reply:
I think `-record_heap` sounds like a good idea, I will make such an option and make the default `-record_function` value as empty.

Reply:
Done